models/scheduled_tasks_helpers_test.go

```diff
@@ -0,0 +1,158 @@
+	db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+	tx, err := db.Begin()
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		require.NoError(t, tx.Rollback())
+		require.NoError(t, sqlDB.Close())
+	})
+
+	createParams := models.CreateScheduledTaskParams{
+		CronExpression: "* * * * *",
+		Type:           models.ScheduledPrintTask,
+		Data: models.ScheduledTaskData{
+			Print: &models.PrintTaskData{
+				Message: "test",
+			},
+		},
+		Disabled: false,
+	}
+
+	t.Run("CreateAndFind", func(t *testing.T) {
+		task, err := models.CreateScheduledTask(tx.Querier, createParams)
+		assert.NoError(t, err)
+
+		task, err = models.FindScheduledTaskByID(tx.Querier, task.ID)
+		assert.NoError(t, err)
+		assert.NotEmpty(t, task.ID)
+		assert.Equal(t, createParams.CronExpression, task.CronExpression)
+		assert.Equal(t, createParams.Type, task.Type)
+		assert.Equal(t, createParams.Disabled, task.Disabled)
+		require.NotNil(t, task.Data.Print)
+		assert.Equal(t, createParams.Data.Print.Message, task.Data.Print.Message)
+
+		task, err = models.CreateScheduledTask(tx.Querier, models.CreateScheduledTaskParams{
+			CronExpression: "a * * * *",
+			Type:           models.ScheduledPrintTask,
+		})
+		require.NotNil(t, err)
+		assert.Contains(t, err.Error(), "Invalid cron expression")
+	})
+
+	t.Run("Change", func(t *testing.T) {
+		task, err := models.CreateScheduledTask(tx.Querier, createParams)
+		assert.NoError(t, err)
+
+		changeParams := models.ChangeScheduledTaskParams{
+			NextRun: pointer.ToTime(time.Now()),
+			LastRun: pointer.ToTime(time.Now()),
+			Disable: pointer.ToBool(true),
+			Running: pointer.ToBool(true),
+			Error:   pointer.ToString("something"),
+		}
+		task, err = models.ChangeScheduledTask(tx.Querier, task.ID, changeParams)
+		assert.NoError(t, err)
+		assert.Equal(t, *changeParams.NextRun, task.NextRun)
+		assert.Equal(t, *changeParams.LastRun, task.LastRun)
+		assert.Equal(t, *changeParams.Disable, task.Disabled)
+		assert.Equal(t, *changeParams.Running, task.Running)
+		assert.Equal(t, *changeParams.Error, task.Error)
+	})
+
+	t.Run("Remove", func(t *testing.T) {
+		task, err := models.CreateScheduledTask(tx.Querier, createParams)
+		assert.NoError(t, err)
+
+		err = models.RemoveScheduledTask(tx.Querier, task.ID)
+		assert.NoError(t, err)
+
+		_, err = models.FindScheduledTaskByID(tx.Querier, task.ID)
+		assert.Error(t, err, "task is not removed")
+	})
+
+	t.Run("Find", func(t *testing.T) {
+		findTX, err := db.Begin()
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			_ = findTX.Rollback()
+		})
+		createParams2 := createParams
+		task1, err := models.CreateScheduledTask(findTX.Querier, createParams2)
+		require.NoError(t, err)
+
+		createParams2.Disabled = true
+		task2, err := models.CreateScheduledTask(findTX.Querier, createParams2)
+		require.NoError(t, err)
+
+		type testCase struct {
+			filter models.ScheduledTasksFilter
+			ids    []string
+		}
+
+		tests := []testCase{
```

Comment:
What about other fields in `ScheduledTasksFilter`?

Reply:
Rigth now there is only one task type (in this PR), so I don't know to test it.

Reply:
You are testing model, so you can pass any value to `Types` even if they are not in our logic. e.g.: `ScheduledTaskType("writesomethingintest")`
Don't forget about `LocationID` and `ServiceID`

Reply:
I can't because there is check for type. So I can't create it by our helper.
I added more tests here https://github.com/percona/pmm-managed/pull/778/commits/d342a83962f2f6df488f9bea9d7aa582f0e2ff6f